core/clients/key_flow.go

```diff
@@ -36,7 +36,7 @@ const (
 type KeyFlow struct {
 	client        *http.Client
 	config        *KeyFlowConfig
-	doer          func(client *http.Client, req *http.Request, cfg *RetryConfig) (resp *http.Response, err error)
+	doer          func(req *http.Request) (resp *http.Response, err error)
```

Comment:
Same thoughts as https://github.com/stackitcloud/stackit-sdk-go/pull/337#discussion_r1503968736, because this matches the signature of `http.RoundTripper.RoundTrip` (but it also matches the signature of `http.Client.Do`)